server/src/test/java/org/opensearch/cluster/service/MasterTaskThrottlerTests.java

```diff
@@ -0,0 +1,352 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import static org.opensearch.test.ClusterServiceUtils.setState;
+
+/**
+ * Contains tests for {@link MasterTaskThrottler}
+ */
+public class MasterTaskThrottlerTests extends OpenSearchTestCase {
+
+    private static ThreadPool threadPool;
+    private ClusterService clusterService;
+    private DiscoveryNode localNode;
+    private DiscoveryNode[] allNodes;
+    private MasterThrottlingStats throttlingStats;
+
+    @BeforeClass
+    public static void beforeClass() {
+        threadPool = new TestThreadPool("TransportMasterNodeActionTests");
+    }
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        clusterService = ClusterServiceUtils.createClusterService(threadPool);
+        localNode = new DiscoveryNode(
+            "local_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.CLUSTER_MANAGER_ROLE),
+            Version.V_3_0_0
+        );
+        allNodes = new DiscoveryNode[] { localNode };
+        throttlingStats = new MasterThrottlingStats();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        clusterService.close();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        threadPool = null;
+    }
+
+    public void testDefaults() {
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.builder().build(), ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        MasterTaskThrottler throttler = new MasterTaskThrottler(
+            clusterSettings,
+            () -> { return clusterService.getMasterService().getMinNodeVersion(); },
+            throttlingStats
+        );
+        for (String key : MasterTaskThrottler.CONFIGURED_TASK_FOR_THROTTLING) {
+            assertNull(throttler.getThrottlingLimit(key));
+        }
+    }
+
+    public void testValidateSettingsForDifferentVersion() {
+        DiscoveryNode masterNode = new DiscoveryNode(
+            "local_master_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.CLUSTER_MANAGER_ROLE),
+            Version.V_3_0_0
+        );
+        DiscoveryNode dataNode = new DiscoveryNode(
+            "local_data_node",
+            buildNewFakeTransportAddress(),
+            Collections.emptyMap(),
+            Collections.singleton(DiscoveryNodeRole.DATA_ROLE),
+            Version.V_1_0_0
+        );
+        setState(clusterService, ClusterStateCreationUtils.state(masterNode, masterNode, new DiscoveryNode[] { masterNode, dataNode }));
```

Comment:
repeated across all test, move them to a private methods

Reply:
Done.